Mathlib/Data/Set/Intervals/UnorderedInterval.lean

```diff
@@ -0,0 +1,390 @@
+#align set.interval_eq_union Set.interval_eq_union
+
+theorem mem_interval : a ∈ [[b, c]] ↔ b ≤ a ∧ a ≤ c ∨ c ≤ a ∧ a ≤ b := by simp [interval_eq_union]
+#align set.mem_interval Set.mem_interval
+
+theorem not_mem_interval_of_lt (ha : c < a) (hb : c < b) : c ∉ [[a, b]] :=
+  not_mem_Icc_of_lt <| lt_min_iff.mpr ⟨ha, hb⟩
+#align set.not_mem_interval_of_lt Set.not_mem_interval_of_lt
+
+theorem not_mem_interval_of_gt (ha : a < c) (hb : b < c) : c ∉ [[a, b]] :=
+  not_mem_Icc_of_gt <| max_lt_iff.mpr ⟨ha, hb⟩
+#align set.not_mem_interval_of_gt Set.not_mem_interval_of_gt
+
+theorem interval_subset_interval_iff_le :
+    [[a₁, b₁]] ⊆ [[a₂, b₂]] ↔ min a₂ b₂ ≤ min a₁ b₁ ∧ max a₁ b₁ ≤ max a₂ b₂ :=
+  interval_subset_interval_iff_le'
+#align set.interval_subset_interval_iff_le Set.interval_subset_interval_iff_le
+
+/-- A sort of triangle inequality. -/
+theorem interval_subset_interval_union_interval : [[a, c]] ⊆ [[a, b]] ∪ [[b, c]] := fun x => by
+  simp only [mem_interval, mem_union]
+  cases' le_total a c with h1 h1 <;>
+  cases' le_total x b with h2 h2 <;>
+  -- Porting note: restore `tauto`
+  aesop
+#align set.interval_subset_interval_union_interval Set.interval_subset_interval_union_interval
+
+theorem monotone_or_antitone_iff_interval :
+    Monotone f ∨ Antitone f ↔ ∀ a b c, c ∈ [[a, b]] → f c ∈ [[f a, f b]] := by
+  constructor
+  · rintro (hf | hf) a b c <;> simp_rw [← Icc_min_max, ← hf.map_min, ← hf.map_max]
+    exacts[fun hc => ⟨hf hc.1, hf hc.2⟩, fun hc => ⟨hf hc.2, hf hc.1⟩]
+  contrapose!
+  rw [not_monotone_not_antitone_iff_exists_le_le]
+  rintro ⟨a, b, c, hab, hbc, ⟨hfab, hfcb⟩ | ⟨hfba, hfbc⟩⟩
+  · exact ⟨a, c, b, Icc_subset_interval ⟨hab, hbc⟩, fun h => h.2.not_lt <| max_lt hfab hfcb⟩
+  · exact ⟨a, c, b, Icc_subset_interval ⟨hab, hbc⟩, fun h => h.1.not_lt <| lt_min hfba hfbc⟩
+#align set.monotone_or_antitone_iff_interval Set.monotone_or_antitone_iff_interval
+
+theorem monotoneOn_or_antitoneOn_iff_interval :
+    MonotoneOn f s ∨ AntitoneOn f s ↔
+      ∀ (a b c) (_ : a ∈ s) (_ : b ∈ s) (_ : c ∈ s), c ∈ [[a, b]] → f c ∈ [[f a, f b]] :=
+  by
+  rw [monotoneOn_iff_monotone, antitoneOn_iff_antitone, monotone_or_antitone_iff_interval]
+  constructor
+  · intros h a' b' c' ha hb hc hc'
+    have := h ⟨a', ha⟩ ⟨b', hb⟩ ⟨c', hc⟩
+    rw [Subtype.coe_mk, Subtype.coe_mk, Subtype.coe_mk] at this
+    apply this
+    rwa [mem_interval, Subtype.mk_le_mk, Subtype.mk_le_mk, Subtype.mk_le_mk, Subtype.mk_le_mk,
+      ← mem_interval]
+  · intros h a' b' c' hc
+    have := h a' b' c' a'.2 b'.2 c'.2
+    apply this
+    rwa [mem_interval, Subtype.mk_le_mk, Subtype.mk_le_mk, Subtype.mk_le_mk, Subtype.mk_le_mk,
+      ← mem_interval] at hc
+  -- Porting note: `simp` can't `rw` under `∀`
+  -- simp [monotoneOn_iff_monotone, antitoneOn_iff_antitone, monotone_or_antitone_iff_interval,
+    -- mem_interval]
+#align set.monotone_on_or_antitone_on_iff_interval Set.monotoneOn_or_antitoneOn_iff_interval
+
+-- Porting note: what should the naming scheme be here?
```

Comment:
Can you clarify? What is the question here? Isn't it just called `interval_oc`? If there's been a discussion elsewhere please link to it.

Reply:
The auto generated name was `intervalOc`, because it is a subset. The current name is closer to the `Ioc` naming scheme, but it would be an exception to the rule

Reply:
https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60interval_oc.60.20naming.20scheme/near/316853344

Reply:
Keep as `interval_oc`, and add a link to the zulip thread.